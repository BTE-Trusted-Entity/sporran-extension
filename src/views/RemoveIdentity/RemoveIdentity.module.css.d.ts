export const __esModule: true;
export const container: string;
export const heading: string;
export const subline: string;
export const name: string;
export const addressLabel: string;
export const address: string;
export const explanation: string;
export const buttonsLine: string;
export const cancel: string;
export const remove: string;
