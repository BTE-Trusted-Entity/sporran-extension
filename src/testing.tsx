export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
