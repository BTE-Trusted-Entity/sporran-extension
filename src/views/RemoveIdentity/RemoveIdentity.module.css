.container {
  composes: container from '../../components/View/View.module.css';
}

.heading {
  composes: screenHeadline from '../../components/Typography/Typography.module.css';
}

.subline {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 0 0 0.8125rem;
}

.name {
  composes: identityText from '../../components/Typography/Typography.module.css';
  margin: 0.5rem 0 0.25rem;
  max-width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.addressLabel {
  composes: smallText from '../../components/Typography/Typography.module.css';
  margin: 0.5rem 0;
}

.address {
  composes: inputText from '../../components/Typography/Typography.module.css';
  margin: 0;
}

.explanation {
  composes: subline from '../../components/Typography/Typography.module.css';
  margin: 1.5rem 0;
}

.buttonsLine {
  composes: buttonsLine from '../../components/Button/Button.module.css';
  margin: 0;
}

.cancel {
  composes: buttonPrimary from '../../components/Button/Button.module.css';
}

.remove {
  composes: buttonSecondary from '../../components/Button/Button.module.css';
  padding: 0 1rem;
}
