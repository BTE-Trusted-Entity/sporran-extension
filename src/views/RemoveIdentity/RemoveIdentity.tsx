import { useCallback } from 'react';
import { browser } from 'webextension-polyfill-ts';
import { Link, useHistory } from 'react-router-dom';
import { map, without } from 'lodash-es';

import * as styles from './RemoveIdentity.module.css';

import { Identity } from '../../utilities/identities/types';
import { Balance } from '../../components/Balance/Balance';
import {
  removeIdentity,
  useIdentities,
} from '../../utilities/identities/identities';
import { LinkBack } from '../../components/LinkBack/LinkBack';
import { Stats } from '../../components/Stats/Stats';
import { generatePath, paths } from '../paths';
import { Avatar } from '../../components/Avatar/Avatar';

interface Props {
  identity: Identity;
}

export function RemoveIdentity({ identity }: Props): JSX.Element {
  const t = browser.i18n.getMessage;

  const history = useHistory();
  const { data: identities } = useIdentities();

  const handleClick = useCallback(async () => {
    if (!identities) {
      return;
    }

    const addresses = map(identities, 'address');
    const remaining = without(addresses, identity.address);
    const firstAddress = remaining[0];
    const path = firstAddress
      ? generatePath(paths.identity.overview, { address: firstAddress })
      : paths.home;
    history.push(path);

    await removeIdentity(identity);
  }, [history, identity, identities]);

  return (
    <main className={styles.container}>
      <h1 className={styles.heading}>{t('view_RemoveIdentity_heading')}</h1>
      <p className={styles.subline}>{t('view_RemoveIdentity_subline')}</p>

      <Avatar identity={identity} />
      <p className={styles.name}>{identity.name}</p>

      <Balance address={identity.address} />

      <small className={styles.addressLabel}>
        {t('view_RemoveIdentity_address')}
      </small>
      <p className={styles.address}>{identity.address}</p>
      <p className={styles.explanation}>
        {t('view_RemoveIdentity_explanation')}
      </p>

      <p className={styles.buttonsLine}>
        <Link to={paths.home} className={styles.cancel}>
          {t('common_action_cancel')}
        </Link>
        <button type="button" onClick={handleClick} className={styles.remove}>
          {t('view_RemoveIdentity_remove')}
        </button>
      </p>

      <LinkBack />

      <Stats />
    </main>
  );
}
