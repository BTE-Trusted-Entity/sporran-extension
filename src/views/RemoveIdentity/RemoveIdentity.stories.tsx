import { Meta } from '@storybook/react';

import { identitiesMock as identities } from '../../utilities/identities/IdentitiesProvider.mock';

import { RemoveIdentity } from './RemoveIdentity';

export default {
  title: 'Views/RemoveIdentity',
  component: RemoveIdentity,
} as Meta;

const identity = identities['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1'];

export function Template(): JSX.Element {
  return <RemoveIdentity identity={identity} />;
}
