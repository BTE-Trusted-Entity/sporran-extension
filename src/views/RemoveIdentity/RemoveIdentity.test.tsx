import { identitiesMock, render } from '../../testing/testing';

import { RemoveIdentity } from './RemoveIdentity';

const identity =
  identitiesMock['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1'];

describe('RemoveIdentity', () => {
  it('should render', () => {
    const { container } = render(<RemoveIdentity identity={identity} />);
    expect(container).toMatchSnapshot();
  });
});
