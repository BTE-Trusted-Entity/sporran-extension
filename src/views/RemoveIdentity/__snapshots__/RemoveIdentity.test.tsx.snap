// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveIdentity should render 1`] = `
<div>
  <main
    class="container"
  >
    <h1
      class="heading"
    >
      Attention!
    </h1>
    <p
      class="subline"
    >
      Youâ€™re about to remove your Identity
    </p>
    <div
      class="avatar"
    >
      Identicon
    </div>
    <p
      class="name"
    >
      Light DID Identity
    </p>
    <p
      class="balanceLine"
    >
      Total Balance: 
      <span
        class="amount"
      >
        1.2340
         
        <span
          aria-label="KILT Coin"
          class="coin"
        />
      </span>
    </p>
    <small
      class="addressLabel"
    >
      My Identity address
    </small>
    <p
      class="address"
    >
      4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1
    </p>
    <p
      class="explanation"
    >
      The next step will remove your Identity from the browser extension. This will not delete your Identity. You can add it again any time by importing your backup phrase.
    </p>
    <p
      class="buttonsLine"
    >
      <a
        class="cancel"
        href="/"
      >
        Cancel
      </a>
      <button
        class="remove"
        type="button"
      >
        Remove this Identity
      </button>
    </p>
    <button
      aria-label="Back"
      class="linkBack"
      title="Back"
      type="button"
    />
    <p
      class="stats"
    >
      You have 3 Identities with a
      <span
        class="balance"
      >
         Total Balance: 
        <span
          class="amount"
        >
          3.7020
           
          <span
            aria-label="KILT Coin"
            class="coin"
          />
        </span>
      </span>
    </p>
  </main>
</div>
`;
